qiskit/pulse/transforms/alignments.py

```diff
@@ -0,0 +1,454 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""A collection of passes to reallocate the timeslots of instructions according to context."""
+
+import abc
+from typing import Optional, Iterable, Callable, Dict, Any
+
+import numpy as np
+
+from qiskit.pulse import channels as chans, instructions
+from qiskit.pulse.exceptions import PulseError
+from qiskit.pulse.schedule import Schedule, ScheduleComponent
+
+
+class AlignmentTransform(abc.ABC):
+    """An abstract class for schedule alignment."""
+    is_sequential = None
+
+    @abc.abstractmethod
+    def align(self, schedule: Schedule) -> Schedule:
```

Comment:
Does this only schedule a schedule, not a schedule block?

Reply:
I think this is only schedule, because this is called by `qiskit.pulse.transforms.canonicalization.block_to_schedule`. So everything is converted into schedule then transformed.